crates/polars-io/src/file_cache/eviction.rs

```diff
@@ -0,0 +1,194 @@
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+
+use fs4::FileExt;
+use polars_core::config;
+use polars_error::PolarsResult;
+
+use super::cache_lock::{GlobalFileCacheGuardExclusive, GLOBAL_FILE_CACHE_LOCK};
+use crate::pl_async;
+
+pub(super) struct EvictionManager {
+    pub(super) prefix: Arc<Path>,
+    pub(super) files_to_remove: Option<(SystemTime, Vec<PathBuf>)>,
+    pub(super) limit_since_last_access: Duration,
+}
+
+impl EvictionManager {
+    pub(super) fn run_in_background(mut self) {
+        let verbose = config::verbose();
+        let sleep_interval = std::cmp::max(self.limit_since_last_access.as_secs() / 4, 7);
```

Comment:
Can we set this higher to 60 seconds or so? Doesn't have to run often I think?

Reply:
updated to 60 secs